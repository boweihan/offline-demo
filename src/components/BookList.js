import React from 'react';
import PropTypes from 'prop-types';
import BookListItem from './BookListItem';
import ListHeading from './ListHeading';

const style = {
  container: {
    flex: 1,
    padding: 10,
  },
  list: {
    listStyle: 'none',
    padding: 0,
    margin: 0,
  },
};

const BookList = ({ heading, books, selectedUser, checkout }) => (
  <div style={style.container}>
    <ListHeading text={heading} />
    <ul style={style.list}>
      {books.map(book => {
        return (
          <BookListItem
            key={book.id}
            {...book}
            checkout={checkout && (() => checkout(book, selectedUser))}
          />
        );
      })}
    </ul>
  </div>
);

BookList.propTypes = {
  heading: PropTypes.string.isRequired,
  books: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      checkedOutBy: PropTypes.obj,
      title: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
  selectedUser: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }),
  checkout: PropTypes.func,
};

export default BookList;
