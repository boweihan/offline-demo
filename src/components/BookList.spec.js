import React from 'react';
import BookList from './BookList';
import renderer from 'react-test-renderer';

describe('BookList component', () => {
  it('deep snapshot renders correctly', () => {
    const props = {
      heading: 'snapshot',
      books: [
        {
          id: 1,
          title: 'a',
        },
        {
          id: 2,
          title: 'b',
        },
      ],
    };

    const tree = renderer.create(<BookList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
