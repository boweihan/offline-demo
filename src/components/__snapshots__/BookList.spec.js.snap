// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList component deep snapshot renders correctly 1`] = `
<div
  style={
    Object {
      "flex": 1,
      "padding": 10,
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "darkgray",
      }
    }
  >
    <h2
      style={
        Object {
          "color": "white",
          "fontFamily": "Open Sans",
          "margin": 5,
          "textAlign": "center",
        }
      }
    >
      snapshot
    </h2>
  </div>
  <ul
    style={
      Object {
        "listStyle": "none",
        "margin": 0,
        "padding": 0,
      }
    }
  >
    <li
      className="list_item"
      style={
        Object {
          "backgroundColor": "#f2f2f2",
          "fontFamily": "Open Sans",
          "margin": "2px 0",
          "padding": 10,
        }
      }
    >
      a
    </li>
    <li
      className="list_item"
      style={
        Object {
          "backgroundColor": "#f2f2f2",
          "fontFamily": "Open Sans",
          "margin": "2px 0",
          "padding": 10,
        }
      }
    >
      b
    </li>
  </ul>
</div>
`;
